storage_service/locations/models/package.py

```diff
@@ -1112,9 +1141,34 @@ def finish_reingest(self, origin_location, origin_path, reingest_location, reing
             distutils.dir_util.copy_tree(reingest_metadata_dir,
                 original_metadata_dir)
 
+        # Replace preservation derivatives
+        reingest_objects_dir = os.path.join(reingest_full_path, 'data', 'objects')
+        original_objects_dir = os.path.join(path, 'data', 'objects')
+        preservation_regex = r'(.+)-\w{8}-\w{4}-\w{4}-\w{4}-\w{12}'
+        # Walk through all files
+        for dirpath, _, filepaths in os.walk(reingest_objects_dir):
+            for filepath in filepaths:
+                match = re.match(preservation_regex, filepath)
+                # If preservation file, copy and delete the old one
+                if match:
+                    reingest_preservation_path = os.path.join(dirpath, filepath)
+                    original_preservation_path = reingest_preservation_path.replace(reingest_objects_dir, original_objects_dir)
+                    # Check for another preservation derivative and delete
+                    dest_dir = os.path.dirname(original_preservation_path)
+                    dupe_preservation_regex = match.group(1) + r'-\w{8}-\w{4}-\w{4}-\w{4}-\w{12}'
+                    for p in os.listdir(dest_dir):
+                        if re.match(dupe_preservation_regex, p):
+                            del_path = os.path.join(dest_dir, p)
+                            LOGGER.info('Deleting %s', del_path)
+                            os.remove(del_path)
+                    # Copy new preservation derivative
+                    LOGGER.info('Moving %s to %s', reingest_preservation_path, original_preservation_path)
+                    shutil.copy2(reingest_preservation_path, original_preservation_path)
+
```

Comment:
Nice catch.  Updated so the regex matches on the extension as well (so originalfile-originaluuid.ext doesn't match both originalfile-originaluuid.ext and originalfile-originaluuid-preservationuuid.ext), and added a check to not delete itself if detected as a duplicate preservation file.  I'll rebase the fix into the original commit if it looks good.